### DISCUSSION TITLE
Change color of background on CheckBox in AVA11

### DISCUSSION THREAD

**Discussion by hayriyedurgun** [AvaloniaUI/Avalonia#14404] [Q&A]
https://github.com/AvaloniaUI/Avalonia/discussions/6679 -> this one didn't work in AVA11

I attempted various approaches, but none seemed to work as intended.

        <Grid>
            <Grid.Styles>
                
                <Style Selector="CheckBox /template/ Grid > Border#NormalRectangle">
                    <Setter Property="Background" Value="Red" />
                </Style>
                
               
                <Style Selector="CheckBox Border#NormalRectangle">
                    <Setter Property="Background" Value="Red" />
                </Style>
            
                
                <Style Selector="CheckBox Border#NormalRectangle">
                    <Setter Property="Background" Value="Red" />
                </Style>
            </Grid.Styles>
    
            <CheckBox Grid.Row="0" Margin="50"/>
        </Grid>

Upon inspecting with Developer Tools, it appeared to be utilizing 'CheckBoxCheckBackgroundFillUnchecked,' overriding other styles.  I experimented with the following:

      <Grid>
          <Grid.Resources>
              <ResourceDictionary>
                  <ResourceDictionary.ThemeDictionaries>
                      <ResourceDictionary x:Key="Default">
                          <SolidColorBrush x:Key="TestBrush" Color="Pink" />
                          <StaticResource x:Key="CheckBoxCheckBackgroundFillUnchecked"
                                          ResourceKey="TestBrush" />
                      </ResourceDictionary>
                  </ResourceDictionary.ThemeDictionaries>
              </ResourceDictionary>
          </Grid.Resources>
  
          <CheckBox Grid.Row="0" Margin="50"/>
      </Grid>

It worked but is there another option to change background color without override all checkbox color resources? I mean, I don't want to override [these](https://github.com/AvaloniaUI/Avalonia/blob/2a9e7e9dcfc3e590bdd50b1e874ce70c0fafcc24/src/Avalonia.Themes.Fluent/Accents/FluentControlResources.xaml#L292) resources

**Comment by timunie** [comment]
check original style

**Reply by timunie** [reply]
you should be able to just override the resources in that style I believe

**Reply by hayriyedurgun** [reply]
the original fluent style is [here](https://github.com/AvaloniaUI/Avalonia/blob/2a9e7e9dcfc3e590bdd50b1e874ce70c0fafcc24/src/Avalonia.Themes.Fluent/Controls/CheckBox.xaml#L38).  

All other styles(Unchecked PointerOver, Unchecked Pressed, Unchecked Disabled, Checked Normal, Checked PointerOver..) can override with this

        <Style Selector="^ /template/ Border#NormalRectangle">
          <Setter Property="BorderBrush" Value="{DynamicResource CheckBoxCheckBackgroundStrokeCheckedPointerOver}" />
          <Setter Property="Background" Value="{DynamicResource CheckBoxCheckBackgroundFillCheckedPointerOver}" />
        </Style>


but I cannot for unchecked normal state.

**Reply by timunie** [reply]
```xml
<Style Selector="CheckBox.My">
    <Style.Resources>
        <SolidColorBrush x:Key="CheckBoxCheckBackgroundStrokeCheckedPointerOver">Red</SolidColorBrush>
        <SolidColorBrush x:Key="CheckBoxCheckBackgroundFillCheckedPointerOver">GreenYellow</SolidColorBrush>
    </Style.Resources>
</Style>
```

My is a special class I introduced. Works also without.

**Reply by tfishr** [reply]
If you need to use Binding for a CheckBox that is inside a DataGridCell and you need to set a custom color for each, you can't override the resources because it will try to bind to the wrong DataContext. If you need to bind to a row, you have to do it by overriding every Selector from the CheckBox.xaml file found in Fluent. Gets complex pretty quickly, and you will need many Converters, but it seems to work.
